File: 2. Fundamentos de Programación OOP/Python/clases.py
# 1. Define una clase `Persona` con atributos `nombre` y `edad`, y un método para mostrar la información.
# 2. Implementa un método estático que cuente la cantidad de instancias creadas de la clase.


class Persona:
    cantidad = 0

    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad
        Persona.cantidad += 1

    def mostrar_info(self):
        print(f"Nombre: {self.nombre}, Edad: {self.edad}")

    @staticmethod
    def contar_instancias():
        print(f"Cantidad de instancias: {Persona.cantidad}")


p1 = Persona("Juan", 30)
p2 = Persona("Ana", 25)
p3 = Persona("Pedro", 35)

p1.mostrar_info()
p2.mostrar_info()
p3.mostrar_info()

Persona.contar_instancias()

File: 2. Fundamentos de Programación OOP/Java/clases.java
// 1. Define una clase `Persona` con atributos `nombre` y `edad`, y un método para mostrar la información.
// 2. Implementa un método estático que cuente la cantidad de instancias creadas de la clase.

class Persona {
    private String nombre;
    private int edad;
    private static int cantidad = 0;

    public Persona(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
        cantidad++;
    }

    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre + ", Edad: " + edad);
    }

    public static int getCantidad() {
        return cantidad;
    }

    public static void main(String[] args) {
        Persona p1 = new Persona("Juan", 20);
        Persona p2 = new Persona("Ana", 25);
        Persona p3 = new Persona("Pedro", 30);

        p1.mostrarInfo();
        p2.mostrarInfo();
        p3.mostrarInfo();

        System.out.println("Cantidad de instancias: " + Persona.getCantidad());
    }
}

